src/Scripting/Core/Script.cs

```diff
@@ -87,10 +95,26 @@ internal static Script<T> CreateInitialScript<T>(ScriptCompiler compiler, string
             ContinueWith<object>(code, options);
 
         /// <summary>
+        /// Continues the script with given <see cref="Stream"/> representing code.
+        /// </summary>
+        /// <exception cref="ArgumentNullException">Stream is null.</exception>
+        /// <exception cref="ArgumentException">Stream is not readable or seekable.</exception>
+        public Script<object> ContinueWith(Stream code, ScriptOptions options = null) =>
+            ContinueWith<object>(code, options);
+
+        /// <summary>
         /// Continues the script with given code snippet.
         /// </summary>
         public Script<TResult> ContinueWith<TResult>(string code, ScriptOptions options = null) =>
-            new Script<TResult>(Compiler, Builder, code ?? "", options ?? InheritOptions(Options), GlobalsType, this);
+            new Script<TResult>(Compiler, Builder, SourceText.From(code ?? "", options?.FileEncoding ?? Options.FileEncoding), options ?? InheritOptions(Options), GlobalsType, this);
+
+        /// <summary>
+        /// Continues the script with given <see cref="Stream"/> representing code.
+        /// </summary>
+        /// <exception cref="ArgumentNullException">Stream is null.</exception>
+        /// <exception cref="ArgumentException">Stream is not readable or seekable.</exception>
+        public Script<TResult> ContinueWith<TResult>(Stream code, ScriptOptions options = null) =>
+            new Script<TResult>(Compiler, Builder, SourceText.From(code, options?.FileEncoding ?? Options.FileEncoding), options ?? InheritOptions(Options), GlobalsType, this);
```

Comment:
>code [](start = 67, length = 4)

check for code == null. Just a detail, but we should be throwing exceptions with the correct parameter name.